src/ProjectAudioManager.cpp

```diff
@@ -728,7 +728,8 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
 
             transportTracks.captureTracks.push_back(newTrack);
          }
-         TrackList::Get( *p ).GroupChannels(*first, recordingChannels);
+         if(recordingChannels == 2)
+            TrackList::Get( *p ).LinkChannels(*first, true);
```

Comment:
Rather than eliminate GroupChannels and replace it with two functions -- preserve GroupChannels.  Keep the argument for number of channels.  The third argument was never non-default -- very well, remove it, but replace with the new argument that specifies whether to align or not, in the case of more than one channel.

Then write the body of GroupChannels on the assumption that the number of channels to be grouped is not necessarily two or less.